src/Nncase.Transform/Rules/Neutral/Fusion.cs

```diff
@@ -30,20 +30,27 @@ public partial class SingleInputFusion<T, BeginT, EndT> : FusionMaker
     where EndT : Op
 {
     /// <inheritdoc/>
-    public override Pattern Pattern { get; } = IsWildcardCall<EndT>("st", null!,
-        IsWildcardCall<T>(null!, null!,
-            IsWildcardCall<BeginT>(null!, null!, IsWildcard("input"))));
-
-    // replace input with var
-    private Call? GetReplace(Call st, Expr input, RunPassOptions passOptions)
+    public override Pattern Pattern { get; } = IsWildcardCall<EndT>("end_call", "end_op",
+        IsWildcardCall<T>("mid_call", "mid_op",
+            IsWildcardCall<BeginT>("begin_call", "begin_op", IsWildcard("input"))));
+
+    protected virtual Call? GetReplace(Call end_call, IReadOnlyList<Expr> end_call_params,
+                             Call mid_call, IReadOnlyList<Expr> mid_call_params,
+                             Call begin_call, IReadOnlyList<Expr> begin_call_params,
+                             Expr input)
     {
-        var arg = new Var("input0", input.CheckedType!);
-        var body = ReplaceTarget(st, input, arg, passOptions.MatchOptions);
-        var fusion = new Call(new Fusion(FullName, ModuleKind, body, new[] { arg }), input);
-        return fusion;
+        var new_input = new Var(input.CheckedType!);
+        var new_begin_call_params = ReplaceParams(begin_call_params, input, new_input);
+        var new_begin_call = begin_call with { Parameters = new(new_begin_call_params) };
 
-        // options.SuppressPattern(st, Pattern);
-        // return fusion;
+        var new_mid_call_params = ReplaceParams(mid_call_params, begin_call, new_begin_call);
```

Comment:
Since a call is reconstructed each time, it is possible to add an implementation of ReplaceParams that replaces the parameters of the call and returns a new call, similar to the previously existing ReplaceParams

Reply:
will add with next pr.